pinecone/index.py

```diff
@@ -56,8 +62,47 @@ def __init__(self, index_name: str, pool_threads=1):
         self._vector_api = VectorOperationsApi(self)
 
     @validate_and_convert_errors
-    def upsert(self, vectors, **kwargs):
+    def upsert(self,
+               vectors: Union[List[Vector], List[Tuple]],
+               namespace: Optional[str] = None,
```

Comment:
This is actually UpsertRequest if you want to be religious about it :)

Reply:
Actually I think it's indeed a list of vectors, the UpsertRequest object is created internally, right?